servicetalk-grpc-api/src/main/java/io/servicetalk/grpc/api/GrpcClientBuilder.java

```diff
@@ -270,11 +270,25 @@
      * <p>
      * This setting disables the default filter such that no {@code Host} header will be manipulated.
      *
+     * @deprecated Use {@link #hostHeaderFallback(boolean)}.
      * @return {@code this}
      * @see #unresolvedAddressToHost(Function)
      */
+    @Deprecated
     public abstract GrpcClientBuilder<U, R> disableHostHeaderFallback();
 
+    /**
+     * Configures automatically setting {@code Host} headers by inferring from the address or {@link HttpMetaData}.
+     * <p>
+     * When {@code false} is passed, this setting disables the default filter such that no {@code Host} header will be
+     * manipulated.
+     *
+     * @param enable Whether a default filter for inferring the {@code Host} headers should be added.
+     * @return {@code this}
+     * @see #unresolvedAddressToHost(Function)
+     */
+    public abstract GrpcClientBuilder<U, R> hostHeaderFallback(boolean enable);
```

Comment:
Adding a new abstract method will be a breaking change for 0.41 branch. Consider adding the default implementation that throws `UnsupportedOperationException` in this PR. After cherry-picking this into 0.41, please submit a follow-up for the `main` branch to remove reprecated methods and make new methods `abstract`. We followed the similar approach here: https://github.com/apple/servicetalk/pull/1781#discussion_r701419264